### ISSUE TITLE
Persistent highlighting of search terms not working

### ISSUE THREAD

**Issue by KatharineV** [scientist-softserv/adventist_knapsack#177]
Testing the Knapsack upgrade, my highlighted search terms do not stay highlighted when I open a work from the catalog search. 

Here's the search where the term is highlighted and shows in a snippet.

![Image](https://github.com/scientist-softserv/adventist-dl/assets/118196520/3b5a4c5c-352c-4e59-9437-edb8bf79aa7e)

Here is the work page that opened with a URL referencing the parent query, but the UV doesn't show my highlighted keyword in the document automatically, which is the behavior I expect. 

https://adl.b2.adventistdigitallibrary.org/concern/published_works/22266468_cardinal_1986?parent_query=swimming
![Image](https://github.com/scientist-softserv/adventist-dl/assets/118196520/b59e50b0-15ec-4545-b3e6-0da0634a80cb)

**Comment by jillpe** [comment]
SoftServ QA: ✅ 

![Image](https://github.com/user-attachments/assets/bd6cd752-bd1f-4f57-b544-86be06a6c1a3)

**Comment by KatharineV** [comment]
Persistent highlighting is not accurate yet. I see two issues. 

**Issue 1**, when I search a single word, the highlighting appears in the UV, but it highlights false matches sometimes. 

**Issue 2**, if I search for more than one word, the persistent highlighting completely fails. This is not an acceptable result, since 99% of our users are searching for multiple words. Almost nobody will search for a single word result in a repository with over 115,000 works. The UV has to know how to deal with multiple search terms. The persistent highlighting shouldn't show in the catalog search for one term out of several keywords and then fail to maintain the highlighting of any of the terms. I know we have a pre-existing (not cutover-related) issue with the UV not being able to search multiple terms. Maybe we need to just turn off persistent highlighting in general if it can't cope with 99% of our users searching for multiple terms. I'm not sure what the value is of fixing the persistent highlighting for single terms (which does need another look from the team) if we can't easily cope with multiple terms. I welcome feedback from the team about this. Maybe we just disable the persistent highlighting if it can't be made functional yet???

**Issue 1 details**
On Staging, I searched for the word "Oakwood" and was directed to a result with "2 results found" for the keyword in the UI. The first instance of the keyword doesn't open automatically (automatic is the expected behavior we used to see--the persistence is meant to open the work at the first instance of the keyword, to create a smooth user experience). When I navigate to the first instance manually, I find that it is NOT the keyword. The UV is highlighting the word "and" even though the keyword "Oakwood" doesn't contain those letters in sequence, and "Oakwood" is in both the URL and in the UV. 

- The work where my keyword appears: 

https://adl.s2.adventistdigitallibrary.org/concern/images/20121910_the_christ_of_the_narrow_way_a_heroic_mural_based_on_the_first_v 

- Work appears as third option in the catalog search for keyword "oakwood" after I sort by Upload Date (I was trying to get a work that ran thru the UV): 

![Image](https://github.com/user-attachments/assets/c9049e55-346e-45f3-b2f1-dbdefb6ad07b)


- UV in the work page says there are two results for "oakwood":

![Image](https://github.com/user-attachments/assets/db28221f-9b88-408b-b5b1-9f53c5cd7db2)


https://github.com/user-attachments/assets/bdf78b29-9b7d-45c5-b77c-59c2a59a1d0f


- First highlighted result returns "and" rather than "oakwood":

![Image](https://github.com/user-attachments/assets/c056e277-7f23-4510-a280-e258915324b8)




**Issue 2 details**
Any search for multiple terms will return good results in the catalog search, highly ranking the most relevant result where the most terms appear. If the work is split by the UV, then you see terms highlighted in the catalog search. However, the UV can't carry the persistence of multiple terms, so you open the work and see nothing. "No matches," it says, which is an inaccurate result that will confuse users. So if we can't fix this, then we should probably get rid of the persistent highlighting. (?)

- Search for term "Twelve men graduated oakwood" returns [the work I know is most relevant](https://adl.s2.adventistdigitallibrary.org/concern/theses/wdf_0003_031__the_contributions_of_twelve_men_who_graduated_from_oakwood_co). It has one of the terms highlighted in the catalog search: 

![Image](https://github.com/user-attachments/assets/a66b6800-89ef-4b2e-9194-f7a74075dbec)


- I open the work and the UV shows me nothing and falsely states the terms don't appear. 
![Image](https://github.com/user-attachments/assets/e4c00635-28b3-4919-a339-8829df4ecd64)

**Comment by KatharineV** [comment]
I tested this in ADL.b2 after everything was switched to PDF.js. I also tested in the Spectrum.b2 and SDAPI.b2 tenants.

In ADL and Spectrum tenants I searched for the phrase "1919 Bible Conference." Both tenants contain the same Spectrum magazine issues, so I expected to be able to test identical works simultaenously. Behavior between the tenants was so vastly different that I had opposite results. 
SDAPI tenant I searched with a different phrase that I knew would return full text, since SDAPI has over 730,000 works and only about 3,000 of them have full text.

**ADL.b2:**
The most relevant results do NOT include full text examples of that phrase where words appear together in full text, even though I know those works appear in the repository. Why did this tenant prioritize works where the term appears in the title? What happened to the full text appearances of the phrase? Is this tenant even searching phrases, or does it ignore the quotation marks?

Search terms not returned as a phrase, see below!

![Image](https://github.com/user-attachments/assets/76afc439-af0f-4e22-b586-042a302efaf0)

Clicking the most relevant result (which had only "Bible" highlighted in the snippet) opens the work page with PDF.js at the top of page 1 of the PDF. No search results are present.. The highlighted keyword from the snippet is NOT persistently highlighted or opened at that word in the viewer, although I know that PDF.js can do those things. See screenshot below.

![Image](https://github.com/user-attachments/assets/70402a96-2b17-4045-b1b3-49aaa834cc35)

**Spectrum.b2:**
The top relevant results DO include the complete phrase in their full text, and the snippets show this phrase in yellow highlight.

Note that the search term is correctly returned as a phrase, see below

![Image](https://github.com/user-attachments/assets/5f5ca745-230a-4ac2-9c6d-a041db8a018f)

When I open the most relevant result, my complete phrase is still highlighted and the PDF.js viewer opens to the first instance of the phrase. This is expected and desired behavior.

![Image](https://github.com/user-attachments/assets/d5fde523-1a84-4b32-8abd-0736bb66db9d)

**SDAPI.b2**
Phrase searching does correctly return multi-word matches highlighted in a snippet. This is desired behavior and good.

![Image](https://github.com/user-attachments/assets/d736eea3-1bba-4295-8bcf-933b7477413c)

However, for some unknown reason (UGH WHY), SDAPI.b2 drops the persistent highlighting when I click into the relevant result. The highlighting is lost in PDF.js, although we know it can retain it in other tenants. See below where the work from above is opened and there is no highlighting and PDF.js opens at the top of the page, as though it retains nothing of my search.

![Image](https://github.com/user-attachments/assets/9ada7d78-5a90-4828-bb2a-aef4a7aef65a)


SUMMARY
Persistent highlighting of search terms, even multi-word phrases (which is desired!!!), works as expected in a new tenant that was created in March 2025. All other tenants lose the persistence at the point of entering the work. ADL.b2 has the worst behavior of all, in that it fails to return the phrase despite the use of quotation marks.

**Comment by KatharineV** [comment]
Tl;dr from above comment

If Notch8 spins up a new Hyku instance for us, we need it to operate like the Spectrum tenant, which is passing the testing criteria for this ticket. 

Can the team figure out why Spectrum.b2 works and the others don't? Can we ensure that the desired behavior (in terms of both multi-word phrase searching AND persistent highlighting in PDF.js) is what we encounter in a new instance?
